### ISSUE TITLE
store encrypted emails with the user's gpg key

### ISSUE THREAD

**Issue by apm256** [pazz/alot#671]
It seems to be impossible, for the moment, to read a sent encrypted email.
It would be great to store it with our own gpg key instead of the addressee's.

**Comment by lucc** [comment]
It seems to me that one has to specify the correct sub key in the gpg.conf file when using the `encrypt-to` option. Using the id of the "parent" key resulted in an error in alot.

**Comment by lamarpavel** [comment]
I have identified a problem with the current approach:

I have two identities: `Anderson` and `Neo`, both have their own key but in the gpg.conf the `encrypt-to` key is set to Neo. Now I am sending a mail as `Anderson` to `Smith`, signing as `Anderson` and encrypting to `Smith`, who is under no circumstances to know of my other identity `Neo`. However, the mail constructed by alot will be encrypted to `Smith` __and__ `Neo`.

There are two ways to solve this as far as I can see:
  1. Don't send and store the exact same mail, allow storing modified versions (eg. encrypted to different keys)
  2. Add an `encrypt-to` option per user account instead of relying on the gpg.conf

**Comment by pazz** [comment]
Quoting lamarpavel (2016-09-24 11:55:55)
>    I have identified a problem with the current approach:
> 
>    I have two identities: Anderson and Neo, both have their own key but in
>    the gpg.conf the encrypt-to key is set to Neo. Now I am sending a mail as
>    Anderson to Smith, signing as Anderson and encrypting to Smith, who is
>    under no circumstances to know of my other identity Neo. However, the mail
>    constructed by alot will be encrypted to Smith and Neo.
> 
>    There are two ways to solve this as far as I can see:
> 
>     1. Don't send and store the exact same mail, allow storing modified
>        versions (eg. encrypted to different keys)
>     2. Add an encrypt-to option per user account instead of relying on the
>        gpg.conf

The second option seems sensible to me as well.

**Comment by meskio** [comment]
I agree the second option sounds better than the first. I personally don't see it as high prio, as I think the sensible thing to do to have several identities that should not be related will be to use different users in the machine. But I think will be a nice feature if someone takes the time to implemented.

**Comment by lamarpavel** [comment]
@meskio There is another use case, which might be relevant to more people: Using gnupg for more than E-Mails.
If, for example, you are using pass[1] to encrypt your passwords and you are using an extra GPG-key for
that, which is not linked to your publicly known GPG-key for E-Mails, then you don't want to auto-encrypt
everything to either of those keys.
I think we, as a broader community, should encourage the use of GPG for different purposes. In that light
it is very inelegant to rely on the GPG config for a feature in alot.

[1]: https://www.passwordstore.org/
